lib/roo_on_rails/checks/heroku/app_exists.rb

```diff
@@ -0,0 +1,69 @@
+require 'roo_on_rails/checks/env_specific'
+require 'roo_on_rails/checks/git/origin'
+require 'roo_on_rails/checks/heroku/token'
+require 'active_support/core_ext/enumerable'
+
+module RooOnRails
+  module Checks
+    module Heroku
+      # Check if a corresponding app exists on Heroku (for a given environment)
+      # 
+      # Input context
+      # - git_repo: the name of the repository
+      # - heroku.api_client: a connected PlatformAPI client
+      # - app_name_stem (optional): a name override
+      #
+      # Output context:
+      # - heroku.app.{env}: an app name.
+      class AppExists < EnvSpecific
+        requires Git::Origin, Heroku::Token
+
+        def intro
+          "Checking if #{bold env} app exist..."
+        end
+
+        def call
+          all_apps = client.app.list.map { |a| a['name'] }
+          matches = all_apps.select { |a| candidates.include?(a) }
+
+          unless matches.one?
+            say "\tcandidates: #{candidates.join(', ')}"
+          end
+
+          if matches.empty?
+            fail! "no apps with matching names detected"
+          end
+
+          if matches.many?
+            fail‼︎ "multiple matching apps detected: #{candidates.map { |c| bold c}.join(', ')}"
+          end
+
+          context.heroku.app![env] = matches.first
+          pass "found app #{bold matches.first}"
+        end
+
+        private
+
+        def name_stem
+          context.app_name_stem || context.git_repo.gsub('.', '')
+        end
+
+        def candidates
+          [
+            [nil, 'roo', 'deliveroo'],
```

Comment:
I'd reverse the order of these so we search for the more specific `deliveroo-` and `roo-` first, to reduce the chance of us detecting somebody else's app.